nixos/modules/services/misc/sr-ht/meta.nix

```diff
@@ -0,0 +1,274 @@
+    # Type u: create user, group if doesn't exist, group name = user name
+    # ID -: allocate automatically
+
+    networking.firewall.allowedTCPPortRanges = [
+      { from = mecfg.port; to = mecfg.port; }
+      # { from = mecfg.debug.port; to = mecfg.debug.port; }
+    ];
+
+    users = {
+      users = [
+        { name = mecfg.user;
+          group = mecfg.user;
+          description = "meta.sr.ht user"; } ];
+      groups = [
+        { name = mecfg.user; } ];
+    };
+
+    services = {
+      # TODO: meta-audit-cleanup ...?
+      # cron.systemCronJobs = [ "0 0 * * * ${meta-sr-ht}/bin/meta-audit-cleanup" ];
+      nginx.virtualHosts."meta.sr.ht" = {
+        # TODO: Figure out whether or not this is actually broken
+        # FIXME: Blocking nixops
+        # enableACME = true;
+        # forceSSL = true;
+        locations = {
+          "/".proxyPass = "http://${cfg.address}:${toString mecfg.port}";
+          "/static".root = "${meta-sr-ht}/${meta-sr-ht.sitePackages}/metasrht";
+        };
+      };
+
+      /*
+      # Need to create postgresql database
+      # Gitlab module snippet
+      # preStart = ''
+      #   if ! test -e "${cfg.statePath}/db-created"; then
+      #     if [ "${cfg.databaseHost}" = "127.0.0.1" ]; then
+      #       ${pkgs.sudo}/bin/sudo -u ${pgSuperUser} psql postgres -c "CREATE ROLE ${cfg.databaseUsername} WITH LOGIN NOCREATEDB NOCREATEROLE ENCRYPTED PASSWORD '${cfg.databasePassword}'"
+      #       ${pkgs.sudo}/bin/sudo -u ${pgSuperUser} ${config.services.postgresql.package}/bin/createdb --owner ${cfg.databaseUsername} ${cfg.databaseName}
+      #       # enable required pg_trgm extension for gitlab
+      #       ${pkgs.sudo}/bin/sudo -u ${pgSuperUser} psql ${cfg.databaseName} -c "CREATE EXTENSION IF NOT EXISTS pg_trgm"
+      #     fi
+      #     ${pkgs.sudo}/bin/sudo -u ${cfg.user} -H ${gitlab-rake}/bin/gitlab-rake db:schema:load
+      #     touch "${cfg.statePath}/db-created"
+      #   fi
+      # '';
+
+      # Might also be broken so FIXME
+      # https://git.sr.ht/~sircmpwn/builds.sr.ht/tree/master/buildsrht/app.py#L8
+      # https://git.sr.ht/~sircmpwn/core.sr.ht/tree/master/srht/database.py
+      preStart = let
+        generateDB = pkgs.writeText "metasrht-gen-db.py" ''
+          #! ${pkgs.python36Packages.python}/bin/python
+          from mansrht import db
+          db.create()
+        '';
+      in ''
+        ${pkgs.python36Packages.python}/bin/python ${generateDB}
+      '';
+      */
+
+      postgresql = {
+        #
+      };
+    };
+
+    systemd.services."meta.sr.ht" = {
+      after = [ "network.target" ];
+      # after = [ "postgresql.service" "network.target" ];
+      # requires = [ "postgresql.service" ];
+      wantedBy = [ "multi-user.target" ];
+
+      description = "meta.sr.ht website service";
+      path = with pkgs; [ python36Packages.gunicorn ];
+      environment.PYTHONPATH = builtins.concatStringsSep ":" (with pkgs.python36Packages; [
```

Comment:
I'm pretty certain this isn't the way I should be doing this, including `meta-sr-ht` in `path` only includes the `/bin` directory.

Reply:
https://github.com/NixOS/nixpkgs/pull/54425#discussion_r250533104